test/servinge2e/kourier/servicemesh_test.go

```diff
@@ -382,22 +382,30 @@ func jwtUnsignedToken(payload map[string]interface{}) (string, error) {
 
 // Convenience method to test requests with tokens, reads the response and returns a closed response and the body bits
 // token can be nil, in which case no Authorization header will be sent
-func jwtHTTPGetRequestBytes(url string, token *string) (*http.Response, []byte, error) {
-	req, err := http.NewRequest("GET", url, nil)
+func jwtHTTPGetRequestBytes(ctx *test.Context, url *url.URL, token *string) (*spoof.Response, error) {
+	tlsConfig := servingTest.TLSClientConfig(context.Background(), ctx.T.Logf, &servingTest.Clients{KubeClient: ctx.Clients.Kube})
+	transport := &http.Transport{TLSClientConfig: tlsConfig}
+
+	sc := &spoof.SpoofingClient{
+		Client: &http.Client{Transport: transport},
+		Logf:   ctx.T.Logf,
+	}
+
+	req, err := http.NewRequest("GET", url.String(), nil)
 	if err != nil {
-		return nil, nil, fmt.Errorf("error creating HTTP GET request: %w", err)
+		return nil, fmt.Errorf("error creating HTTP GET request: %w", err)
 	}
 	if token != nil {
 		req.Header.Add("Authorization", "Bearer "+*token)
 	}
 
-	resp, err := http.DefaultClient.Do(req)
+	// Poll, as it is expected OpenShift Router will return 503s until it reconciles the Route.
+	resp, err := sc.Poll(req, spoof.IsOneOfStatusCodes(http.StatusOK, http.StatusUnauthorized, http.StatusForbidden))
```

Comment:
I wonder if we should patch the retry functionality in the s-o repo as well and thus enable us to not poll here (and automatically catch all the other errors, like the cert retries).

Reply:
Ah I see, I thought we did this for retriability, nevermind then!